KarhooUISDK/Routing/BookingFlowCoordinator.swift

```diff
@@ -0,0 +1,159 @@
+//
+//  BookingFlowCoordinator.swift
+//  KarhooUISDK
+//
+//  Created by Diana Petrea on 04.04.2023.
+//  Copyright © 2023 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+import KarhooSDK
+
+final class BookingFlowCoordinator: KarhooUISDKSceneCoordinator {
+    var baseViewController: BaseViewController
+    var navigationController: UINavigationController?
+    var childCoordinators: [KarhooUISDKSceneCoordinator] = []
+    
+    private var bookingCompletion: ((BookingScreenResult) -> Void)?
+    
+    static let shared = BookingFlowCoordinator()
+    
+    private init() {
+        baseViewController = KarhooRidePlanningViewController()
+    }
+    
+    func start(
+        with journeyInfo: JourneyInfo? = nil,
+        passengerDetails: PassengerDetails? = nil,
+        filters: [QuoteListFilter]? = nil,
+        bookingMetadata: [String: Any]? = nil,
+        callback: ((BookingScreenResult) -> Void)?
+    ) {
+        bookingCompletion = callback
+        
+        saveInjectedData(
+            journeyInfo: journeyInfo,
+            passengerDetails: passengerDetails,
+            filters: filters,
+            bookingMetadata: bookingMetadata
+        )
+        routeToRidePlanning()
+    }
+    
+    private func saveInjectedData(
```

Comment:
Maybe more like "set injected data as current state"? The real feature of the method body is to make sure our data handlers have the same data that was passed to the method.

Reply:
What do you think of `saveInjectedDataForCurrentBooking`?